src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1702,24 +1702,23 @@ public final void stop() {
     public void interrupt() {
         if (this != Thread.currentThread()) {
             checkAccess();
-
+        }
+        // Write interrupted before reading nioBlocker for correct synchronization.
+        interrupted = true;
+        interrupt0();  // inform VM of interrupt
+        if (this != Thread.currentThread()) {
             // thread may be blocked in an I/O operation
```

Comment:
I think the existing comment "thread may be blocked in an I/O operation" can move to before the `if` statement so that it's a bit clearer than this is code for I/O operations.